### PR TITLE
Feature/fqcn

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves FQCN dependency issue in core.

Requires core to be fixed first (CS PR after all major PRs are merged to avoid conflicts).

After this 0.6 ( https://github.com/spryker/code-sniffer/releases ) can be tagged.

### CODE REVIEW

File: Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php [hunk 1]
```diff
@@ -518,4 +696,119 @@ protected function generateUseStatement(array $useStatement)
...
+     *
+     * @throws \RuntimeException
+     *
+     * @return array
+     */
+    protected function parse(PHP_CodeSniffer_File $phpcsFile, $startIndex, $endIndex)
+    {
+        $tokens = $phpcsFile->getTokens();
+        if (empty($tokens[$endIndex])) {
+            throw new RuntimeException('Should not happen');
```
Comment (ThomasKuehnel) [inline Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php]: It would be good to be more specific about the exception. This might apply for all following exceptions.

File: Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php [hunk 2]
```diff
@@ -518,4 +696,119 @@ protected function generateUseStatement(array $useStatement)
...
+    /**
+     * @param \PHP_CodeSniffer_File $phpcsFile
+     * @param int $startIndex
+     * @param int $endIndex
+     *
+     * @throws \RuntimeException
+     *
+     * @return array
+     */
+    protected function parse(PHP_CodeSniffer_File $phpcsFile, $startIndex, $endIndex)
```
Comment (ThomasKuehnel) [inline Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php]: Is there any chance we can split this method up a bit? It's hard to follow all these execution paths.

File: Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php [hunk 3]
```diff
@@ -95,11 +126,75 @@ protected function checkUseForClass(\PHP_CodeSniffer_File $phpcsFile, $stackPtr)
...
+
+        $partial = strpos($statement['content'], '\\') !== 0;
+
+        $extractedUseStatement = ltrim($statement['content'], '\\');
+        $className = substr($statement['content'], strrpos($statement['content'], '\\') + 1);
+
+        $error = 'Use statement ' . $extractedUseStatement . ' for class ' . $className . ' should be in use block.';
+        if ($partial) {
+            // For now just warn about partial FQCN
+            //$phpcsFile->addWarning($error, $stackPtr, 'Interface');
```
Comment (dereuromark) [inline Spryker/Sniffs/Namespaces/SprykerUseStatementSniff.php]: This is more a todo for the near future I hope.
